detailed_docs/Wiki/0249_capabilities.md

```diff
@@ -0,0 +1,135 @@
+## Feature 1: As a mobile application, I want to get HMI capabilities from SDL after Ignition OFF/ON cycle so that I can interact with the HMI efficiently
+
+**Background**
+
+**Given** App requests HMI Capabilities with one of the following RPC
+  * UI.GetSupportedLanguages
+  * UI.GetCapabilities
+  * VR.GetSupportedLanguages
+  * VR.GetCapabilities
+  * TTS.GetSupportedLanguages
+  * TTS.GetCapabilities
+  * Buttons.GetCapabilities
+  * VehicleInfo.GetVehicleType
+  * RC.GetCapabilities
+  * UI.GetLanguage
+  * VR.GetLanguage
+  * TTS.GetLanguage
+#### Scenario 1: SDL Provides HMI capabilities persisted in `HMICapabilitiesCacheFile`
+
+**Given** App registers after Ignition OFF/ON cycle
+
+**When** App requests HMI Capabilities
+
+**Then** SDL checks `HMICapabilitiesCacheFile` in `AppStorageFolder` in [smartDeviceLink.ini](https://github.com/smartdevicelink/sdl_core/blob/master/src/appMain/smartDeviceLink.ini)  
+**And** SDL provides HMI Capabilities persisted in `HMICapabilitiesCacheFile` in the corresponding response.
+
+---
+
+#### Scenario 2: SDL creates `HMICapabilitiesCacheFile` in case it is not present in `AppStorageFolder` in _SmartDeviceLink.ini_ 
+
+**When** SDL checks `HMICapabilitiesCacheFile` and variable is null/empty in [smartDeviceLink.ini](https://github.com/smartdevicelink/sdl_core/blob/master/src/appMain/smartDeviceLink.ini)  
+
+**Then** SDL must create `HMICapabilitiesCacheFile` file  
+
+---
+
+#### Scenario 3: SDL requests HMI capabilities missing in `HMICapabilitiesCacheFile` from HMI
+
+**When** Any capability requested by an app is missing in `HMICapabilitiesCacheFile`
+
+**Then** SDL must default back to `hmi_capabilities.json` file for corresponding capability  
+**And** SDL should send requests to get these capabilities from HMI
```

Comment:
writing this was inspired by
https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0249-Persisting-HMI-Capabilities-specific-to-headunit.md#functional-changes 
> During boot up, SDL Core should check if HMICapabilitiesCacheFile file is present in AppStorageFolder from smartDeviceLink.ini
If present, SDL Core loads the contents of the file and uses that as current system HMI capabilities.
**In case any capability is missing, SDL Core should default back to hmi_capabilities.json file for corresponding capability and should send requests to get these capabilities from HMI.**

maybe, I misunderstood the proposal. please, prompt what I should correct according to implementation?